src/main/java/hudson/plugins/gradle/injection/InjectionUtil.java

```diff
@@ -78,4 +87,49 @@ private static boolean isEnabled(Set<String> labels, Set<String> enabledNodes) {
         }
         return enabledNodes.stream().anyMatch(labels::contains);
     }
+
+    public static boolean isVcsRepositoryAllowed(Run<?, ?> run, InjectionConfig config) {
+        if (config.getParsedInjectionVcsRepositoryPatterns() == null || config.getParsedInjectionVcsRepositoryPatterns().isEmpty()) {
+            return true;
+        }
+
+        if (run instanceof WorkflowRun) {
+            Collection<? extends SCM> scms = ((WorkflowRun) run).getParent().getSCMs();
+
+            for (SCM scm : scms) {
+                if (vcsRepositoryUrlMatches(config, scm)) {
```

Comment:
💭 I guess here we're in a unpredictable case, at least without first experimenting, I would be curious to have some tests, for example a pipeline like:
- checkout not matching
- build maven
- checkout matching
- build maven

Reply:
I did this test manually and injection was applied, but I can also document it as an automated test.